Sources/MediaCategories/MediaCategoryViewController.swift

```diff
@@ -443,16 +443,23 @@ private extension VLCMediaCategoryViewController {
 
 extension VLCMediaCategoryViewController {
 
-    func play(media: VLCMLMedia) {
-        VLCPlaybackController.sharedInstance().fullscreenSessionRequested = media.subtype() != .albumTrack
-        if let collectionModel = model as? CollectionModel, collectionModel.mediaCollection is VLCMLPlaylist || collectionModel.mediaCollection is VLCMLAlbum {
-            guard let index = collectionModel.files.firstIndex(of: media) else {
-                return
-            }
-            VLCPlaybackController.sharedInstance().playMedia(at: index, fromCollection: collectionModel.files)
+    func play(media: VLCMLMedia, at indexPath: IndexPath) {
+        let playbackController = VLCPlaybackController.sharedInstance()
+        let autoPlayNextItem = UserDefaults.standard.bool(forKey: kVLCAutomaticallyPlayNextItem)
+
+        guard autoPlayNextItem else {
+            playbackController.play(media)
+            return
```

Comment:
I find this use of guard a little bit unsettling here. Guard is a keyword that I link to faulty conditions that we prepare against like ensuring something is not nil or that text is not just "" that we want to return from.
I'd be happier if it's just an if. 
It also took me quite a while to understand the code because of it

Reply:
I will make the change!